plutus-tx-plugin/src/PlutusTx/Compiler/Expr.hs

```diff
@@ -793,6 +799,31 @@ compileExpr e = traceCompilation 2 ("Compiling expr:" GHC.<+> GHC.ppr e) $ do
     GHC.Lam b@(GHC.isTyVar -> True) body -> mkTyAbsScoped b $ compileExpr body
     -- otherwise it's a normal lambda
     GHC.Lam b body -> mkLamAbsScoped b $ compileExpr body
+    -- Handle the following case:
```

Comment:
I think we could just roll this into the case below? We would just always use `withVarTyScoped`, and have something like
```
let isWeirdBytestringIntegerLiteral = GHC.eqType (GHC.varType b) GHC.byteArrayPrimTy && isNumericLiteral rhs
ty <- if isWeirdBytestringIntegerLiteral then pure integerTy else compileType ghcType
withVarTyScoped b ty $ ...
```

Reply:
Yeah, good point.